### PR TITLE
Add WebURL and VisibilityLevel to struct

### PR DESCRIPTION
**Pull request by martindekov**
Adding two fields to the GitLab SDK to be vendored
in the GitLab functions

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>

## Description

Two fields added to the structures in the SDK to be released/re-vendored into my code which are for private repositories and whole URL

## How Has This Been Tested?

N.A. It is a struct

## How are existing users impacted? What migration steps/scripts do we need?

N.A.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: sdk/events_scm.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ type GitLabProject struct {
 	Namespace         string `json:"namespace"`
 	Name              string `json:"name"`
 	PathWithNamespace string `json:"path_with_namespace"` //would be repo full name
+	WebURL            string `json:"web_url"`
```
Comment (alexellis) [inline sdk/events_scm.go]: Didn't we settle for `url` or `repo_url`?

Comment (alexellis) [reply sdk/events_scm.go]: `web_url` seems to be right for GitLab. I'm seeing both a `repository` and `project` JSON element in the push event. Which do we want to be using?

```
{
  "object_kind": "push",
  "event_name": "push",
  "before": "e338592127a3aaba36b1e75d21966aecef3dff3e",
  "after": "ecc6e7268563f327577fbe314142b060cf3a425b",
  "ref": "refs/heads/master",
  "checkout_sha": "ecc6e7268563f327577fbe314142b060cf3a425b",
  "message": null,
  "user_id": 12,
  "user_name": "Martin Dekov",
  "user_username": "martindekov",
  "user_email": "mdekov@vmware.com",
  "user_avatar": "https://secure.gravatar.com/avatar/ef817b763c3a2de131d7ea2c92af0afc?s=80&d=identicon",
  "project_id": 7,
  "project": {
    "id": 7,
    "name": "gitlab-tag-event",
    "description": "This is for testing the Tag event if resembles the system hook",
    "web_url": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event",
    "avatar_url": null,
    "git_ssh_url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
    "git_http_url": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event.git",
    "namespace": "martindekov",
    "visibility_level": 0,
    "path_with_namespace": "martindekov/gitlab-tag-event",
    "default_branch": "master",
    "ci_config_path": null,
    "homepage": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event",
    "url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
    "ssh_url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
    "http_url": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event.git"
  },
  "commits": [
    {
      "id": "ecc6e7268563f327577fbe314142b060cf3a425b",
      "message": "Update handler.go",
      "timestamp": "2018-10-05T12:52:28Z",
      "url": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event/commit/ecc6e7268563f327577fbe314142b060cf3a425b",
      "author": {
        "name": "Martin Dekov",
        "email": "mdekov@vmware.com"
      },
      "added": [],
      "modified": [
        "gitlab-tag-event/handler.go"
      ],
      "removed": []
    }
  ],
  "total_commits_count": 1,
  "repository": {
    "name": "gitlab-tag-event",
    "url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
    "description": "This is for testing the Tag event if resembles the system hook",
    "homepage": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event",
    "git_http_url": "https://gitlab.team-serverless.xyz/martindekov/gitlab-tag-event.git",
    "git_ssh_url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
    "visibility_level": 0
  }
}

```

File: sdk/events_scm.go [hunk 2]
```diff
@@ -47,6 +47,8 @@ type GitLabProject struct {
...
+	VisibilityLevel   int    `json:"visibility_level"`
```
Comment (ivanayov) [inline sdk/events_scm.go]: How is this different for GitLab and GitHub?

Comment (alexellis) [reply sdk/events_scm.go]: This doesn't exist in GitHub, it's private true/false.

File: sdk/events_scm.go (hunk 1 above)
Comment (martindekov) [reply sdk/events_scm.go]: `web_url` is always the `https://gitlab.example.com/repo/something.git` the others change depending on the situation like
```
    "url": "git@gitlab.team-serverless.xyz:martindekov/gitlab-tag-event.git",
```
ssh ^

**Pull request merged**